runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java

```diff
@@ -760,4 +769,114 @@ private void saveClientBoundsConfig()
 			configManager.setConfiguration(CONFIG_GROUP, CONFIG_CLIENT_BOUNDS, frame.getBounds());
 		}
 	}
+
+	public void sendCustomNotification(
+		final String title,
+		final String message,
+		final TrayIcon.MessageType type,
+		final String subtitle)
+	{
+		final int width = 360;
+		final int height = 100;
+		final int padding = 50;
+
+		Rectangle display = frame.getGraphicsConfiguration().getBounds();
+
+		JFrame window = new JFrame();
```

Comment:
I still think this should be separate class. Also, probably using JDialog instead of JFrame would make more sense here.

Reply:
Using JDialog is incorrect because a modal notification would be terrible.
As I said on discord, making this a separate class would just be a ctor with this method, because extending JFrame would expose the possibly invalid JFrame.

Reply:
Yes exaclty, just like the SwingUtil class works. I mainly disagree with the amount of code in this method as I said earlier. Even having new Notification(frame) would be still better than having this lying around either in Notifier or ClientUI

Reply:
Also this will now expose 2 public apis for creating notifications, that otherwise would not need to exist.

Reply:
Yes. That sounds correct. Look at the new notification as new reusable component, having it separate is correct. And, the method in ClientUI as factory for displaying this component for anyone who would want to d it.

Reply:
So its okay if we expose a notification that can be misused, but not okay if we expose the main jframe that users won't notice if its misused?

Reply:
Yes. The users not notificing if the jframe is misused is debatable, because you still can change all it's properties from outside when it will be exposed. Exposing the Notification instead of entire JFrame will result in less dumb PRs.

Reply:
I still disagree, but it has been done.